src/main/java/org/broadinstitute/consent/http/service/LibraryCardService.java

```diff
@@ -124,4 +128,62 @@ private void throwIfNull(LibraryCard libraryCard) {
             throw new NotFoundException("LibraryCard not found.");
         }
     }
+
+    //helper method for create method, checks to see if card already exists
+    private void checkIfCardExists(LibraryCard payload) {
+        Integer userId = payload.getUserId();
+        String email = payload.getUserEmail();
+        Integer institutionId = payload.getInstitutionId();
+        Optional<LibraryCard> foundCard;
+        List<LibraryCard> results;
+
+        if(Objects.nonNull(payload.getUserId())) {
+            results = libraryCardDAO.findLibraryCardsByUserId(userId);
+        } else if(Objects.nonNull(email)) {
+            results = libraryCardDAO.findAllLibraryCardsByUserEmail(email);
+        }
+        else {
+            throw new BadRequestException();
+        }
+        foundCard = results.stream().filter(card -> {
+            Boolean sameUserId = Objects.nonNull(userId) && card.getUserId().equals(userId);
+            Boolean sameUserEmail = Objects.nonNull(email) && card.getUserEmail().equalsIgnoreCase(email);
+            Boolean sameInstitution = card.getInstitutionId().equals(institutionId);
+            return (sameUserId || sameUserEmail) && sameInstitution;
+        }).findFirst();
+
+        if(foundCard.isPresent()) {
+            //This exception is a placeholder, what's an exception that can be translated as "CONFLICT"
```

Comment:
We don't have one that directly expresses that, but we sure could use one. Look at `NotFoundException` which extends `ClientErrorException`. We could write our own `ConflictException` and extend `ClientErrorException` as well, and then handle that explicitly in `Resource` where we add dispatchers for known exceptions.